github/VisualStudio
# Ensure only one binding path for extension

Check that only one binding path is active and warn when extra binding paths might cause issues.

### What this PR does

- Find list of `BindingPaths` in Visual Studio configuration using `ShellSettingsManager`
- Check for duplicate binding paths that could load `GitHub.VisualStudio.dll`
- Show warning dialog when extra binding paths might cause issues
- Offer instructions how to resolve this issue (link [here](https://github.com/github/VisualStudio/issues/2006))
- Only execute on `DEBUG` builds (no need to run in production)

### How to test

1. Install GitHub for Visual Studio from Marketplace
2. Load GitHub for Visual Studio solution
3. Launch and install extension
4. Click `View > Other Windows > GitHub`
5. Expect the following to appear in the log
![image](https://user-images.githubusercontent.com/11719160/47292066-d615df80-d5fd-11e8-8748-c7304b5d7adb.png)
6. Click `Yes` and follow instructions
7. Repeat from 2. and check that no dialog appears

### xmldoc

https://github.com/github/VisualStudio/blob/7d5631b06135847377c4eae16d2e3ebf2e189395/src/GitHub.Exports/Helpers/BindingPathUtilities.cs#L10-L32

Fixes #1995
